content/chainguard/chainguard-images/getting-started/nemo/index.md

```diff
@@ -0,0 +1,3013 @@
+  docs:
+    parent: "getting-started"
+weight: 612
+toc: true
+---
+
+[NVIDIA NeMo](https://github.com/NVIDIA/NeMo) is a deep learning framework for building conversational AI models that provides standalone module collections for Automatic Speech Recognition (ASR), Natural Language Processing (NLP), and Text-to-Speech (TTS) tasks. The [NeMo Chainguard Image](/chainguard/chainguard-images/reference/nemo/) is a comparatively lightweight NeMo environment with low to no CVEs, making it ideal for both training and production inference. The NeMo Chainguard Image is designed to work with the [CUDA 12](https://developer.nvidia.com/about-cuda) parallel computing platform, and is suited to workloads that take advantage of connected GPU.
+
+{{< details "What is Deep Learning?" >}}
+{{< blurb/deep-learning >}}
+{{< /details >}}
+
+In this getting started guide, we will use the NeMo Chainguard Image to generate speech from plain text using models provided by NeMo's text-to-speech (TTS) and natural language processing (NLP) collections. In doing so, we'll compare the security and footprint of the NeMo Chainguard Image to the official runtime image distributed by NVIDIA and consider further approaches and resources for applying the NeMo Chainguard Image to additional tasks in conversational AI.
+
+This guide is primarily designed for use in an environment with access to one or more NVIDIA GPUs. However, NVIDIA NeMo is built on [PyTorch Lightning](https://lightning.ai/docs/pytorch/stable/), which supports a wide variety of [accelerators](https://pytorch-lightning.readthedocs.io/en/1.1.8/accelerators.html), or interfaces to categories of processing units (CPU, GPU, TPU) or high-level clustering mechanisms such as [Distributed Data Parallel](https://pytorch.org/tutorials/intermediate/ddp_tutorial.html). Some consideration will be given to alternative computing environments such as CPU in this tutorial.
+
+## Testing Access to GPUs
+
+In our first step, we'll run the NeMo Chainguard Image interactively and determine whether the environment has access to connected GPUs.
+
+If Docker Engine (or Docker Desktop) is not already installed, follow the [instructions for installing Docker Engine on your host machine](https://docs.docker.com/engine/install/). 
+
+Use the below command to pull the image, run it with GPU access, and start a Python interpreter inside the running container.
+
+```bash
+docker run -it --rm \
+  --gpus all \
+  --shm-size=8g \
+  --ulimit memlock=-1 \
+  --ulimit stack=67108864 \
+  cgr.dev/chainguard/nemo:latest
+```
+
+These options allow access to connected GPU, allocate more shared memory to the container, and put an upper bound on container memory use. 
+
+Running the above for the first time may take a few minutes to pull the NeMo Chainguard Image, currently 4.05 GB. Once the image runs, you will be interacting with a bash shell in the running container. Enter the following commands at the prompt to check the availability of your GPU.
```

Comment:
I agree that it would be good to include this, though getting CUDA to work on a local machine is (very) nontrivial. Mac OS is not supported at all if that's the OS you're using. I've written a new blurb that will be included in a prerequisites section. It links to guides on setting  up CUDA for Linux, Windows, GCP, and AWS. As a practical matter I recommend testing on GCP, it's pretty uphill getting this working without preselecting your hardware, OS, and so on for this task. I have updated [this page](https://github.com/chainguard-images/images/blob/main/images/pytorch-cuda12/TESTING.md) with instructions for setting up GCP.

If you were using Mac OS, it looks like that isn't supported at all. :(

Here's the blurb:

<p>
  <a href="https://developer.nvidia.com/cuda-toolkit">Compute Unified Device Architecture (CUDA)</a> is a parallel computing platform developed by NVIDIA. To take advantage of connected NVIDIA GPUs, you’ll need to follow the setup instructions for your local machine or create a CUDA-enabled instance on a cloud provider.
</p>
<p>
  To set up CUDA on your local machine, follow the installation instructions for <a href="https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html">Linux</a> or <a href="https://docs.nvidia.com/cuda/cuda-installation-guide-microsoft-windows/index.html">Windows</a>. CUDA is not currently supported on Mac OS.
</p>
<p>
  Google Cloud Platform provides CUDA-ready <a href="https://cloud.google.com/deep-learning-vm">deep learning instances</a>, including <a href="https://cloud.google.com/deep-learning-vm/docs/pytorch_start_instance">PyTorch-specific instructions</a>. Amazon Web Services also provides CUDA-ready <a href="https://aws.amazon.com/machine-learning/amis/">deep learning instances</a>.
</p>

Reply:
Thank you for adding this section! As someone totally newbie to this area, I did not know things wouldn't work out of the box, so I think it's better to make sure users are aware of everything needed to test the image. I am on Linux, going now through the installation instructions to try to run it locally.

Reply:
I was able to run it on GCP 🎉